### DISCUSSION TITLE
How to consider absolute positioned element as part of floating element bounding box?

### DISCUSSION THREAD

**Discussion by alexbu92** [floating-ui/floating-ui#2746] [Q&A] [answered]
In the floating element I have an absolute positioned element which isn't getting considered when calculating overflow. I can't figure out from the docs how to either include that element's rect or to programmatically compensate by elongating the virtual box.

I've tried adding hardcoded padding in the flip middleware and doing the same in the size middleware to no avail.

**Comment by atomiks** [comment]
Padding in `flip`/`size` (the difference between the layout rect and visual rect)  works for collision detection specifically, although it still gets positioned according to the layout box. 

https://codesandbox.io/p/sandbox/floating-ui-elongated-collision-detection-9322t3?file=%2Fsrc%2FApp.js%3A41%2C36 (resize upward)

Do you have an reproduction?

**Reply by alexbu92** [reply]
Huh that's pretty much what I was trying then but it doesn't work. This is my full UseFloatingOptions:

```typescript
const data = useFloating({
    placement: `bottom-start`,
    middleware: [
      offset(({ rects }) => {
        return -rects.reference.height
      }),
      flip({
        fallbackAxisSideDirection: `start`,
        padding: { bottom: 300 },
      }),
      size({
        apply({ availableHeight, elements }) {
          elements.floating.style.width = elements.reference.getBoundingClientRect().width + `px`
          elements.floating.style.minHeight = elements.reference.getBoundingClientRect().height + `px`
        },
      }),
    ],
    whileElementsMounted: autoUpdate,
  })
  ```
  
  I'll try ripping out a reproducible sandbox.

**Reply by atomiks** [reply, marked as answer]
I think the issue here is that `flip` won't do anything since your code is aligning the floating element directly over the reference element, and flipping depends on the layout rect in relation to the reference element. 

Maybe you want `shift({crossAxis: true, padding: ...})` instead of `flip`? Full demo including styles would help though. 

Also, you can do this:

```js
        apply({ rects, elements }) {
          elements.floating.style.width = rects.reference.width + `px`
          elements.floating.style.minHeight = rects.reference.height + `px`
        },
```
